opentelemetry/proto/common/v1/common.proto

```diff
@@ -21,30 +21,37 @@ option java_package = "io.opentelemetry.proto.common.v1";
 option java_outer_classname = "CommonProto";
 option go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1";
 
+// AnyValue is used to represent any type of attribute value. AnyValue may contain a
+// primitive value such as a string or integer or it may contain an arbitrary nested
+// object containing arrays, key-value lists and primitives.
+message AnyValue {
+  // The value is one of the listed fields. It is valid for all values to be unspecified
+  // in which case this AnyValue is considered to be "null".
+  oneof value {
+    string string_value = 1;
+    bool bool_value = 2;
+    int64 int_value = 3;
+    double double_value = 4;
+    ArrayValue array_value = 5;
+    AttributeKeyValueList kvlist_value = 6;
+  }
+}
+
+message ArrayValue {
+  // Array of values. The array may be empty (contain 0 elements).
+  repeated AnyValue values = 1;
+}
+
+message AttributeKeyValueList {
```

Comment:
I'm somewhat bothered by the Attribute prefix here - do we intend to reuse the term "attribute" for every instance of key-value data? E.g. a field in the log entry, will it also be called "attribute"?

If not, then this type should be simply called KeyValueList and the next one KeyValue, to match AnyValue.

Reply:
+1 

Great point! Let's name it `KeyValueList`.

Reply:
I am guessing renaming AttrributeKeyValue->KeyValue will be a bigger deal since it's an existing name.

Reply:
This was annoying me too a bit. I can rename to KeyValue, no problem with that. We are breaking the proto anyway.

The only reason I was reluctant to use KeyValue was because we also have StringKeyValue. So we will have KeyValue which is a String:AnyValue pair and StringKeyValue which is a String:String pair.

Does this look OK to you?

I would not want to name it KeyAnyValue, it does not looks nice to me.

Reply:
KeyValue and StringKeyValue sgtm 👍

Reply:
OK, changing to KeyValue and KeyValueList.

Reply:
Done.